javascript/javascript3/week2/Rewrite Time/rewrite.js

```diff
@@ -0,0 +1,35 @@
+// REWRITE
+// SetTimeout a promise
+const promiseTimeOut = () => {
+    const promise = new Promise((resolve, reject) => {
+            setTimeout(() => {
+                resolve();
+            }, 3000);
+        })
+        .then(() => console.log("Called after 3 seconds"))
+        .catch((error) => console.error(error));
+};
+promiseTimeOut();
+
+// Get position
+let lat;
```

Comment:
Why declare these variables outside?
We ususally try to have our functions return something. That we we don't have to do all the work inside it.
I assume you mean something like this:

```
function getCurrentLocation() {

  if (navigator.geolocation) {
    return new Promise((resolve, reject) => {
      try {
        navigator.geolocation.getCurrentPosition((position) => {
          resolve(
            {
                long : position.coords.longitude.toFixed(2),
                lat : position.coords.latitude.toFixed(2)
            });
        });
      } catch (error) {
        reject(error);
      }
    });
  }
}

async function displayPositionInMap({ lat, long }) {
  const map = await fetch(
    `https://www.openstreetmap.org/#map=18/${lat}/${long}`
  );
  const mapdata = await map.json();
  return mapData;
}


(async () => {
    const longAndLat = await getCurrentLocation();
    const mapData = await displayPositionInMap(longAndLat)
})();
```

I still get a cors error though